drivers/spi/spi_gd32.c

```diff
@@ -195,6 +253,115 @@ static int spi_gd32_frame_exchange(const struct device *dev)
 	return spi_gd32_get_err(cfg);
 }
 
+#ifdef CONFIG_SPI_GD32_DMA
+static void spi_gd32_dma_callback(const struct device *dmadev, void *arg,
+				  uint32_t channel, int status);
+
+static uint32_t spi_gd32_dma_setup(const struct device *dev, uint32_t dir)
+{
+	const struct spi_gd32_config *cfg = dev->config;
+	struct spi_gd32_data *data = dev->data;
+	const uint32_t datareg = (uint32_t)&SPI_DATA(cfg->reg);
```

Comment:
Suggestion: Remove `datareg`, asign `SPI_DATA` address directly.

Reply:
Applied your suggestion, Thank you.